Kickstarter-iOS/Views/RewardCardContainerViewTests.swift

```diff
@@ -0,0 +1,248 @@
+import Foundation
+@testable import Kickstarter_Framework
+@testable import KsApi
+@testable import Library
+import Prelude
+import UIKit
+
+final class RewardCardContainerViewTests: TestCase {
+  override func setUp() {
+    super.setUp()
+
+    AppEnvironment.pushEnvironment(mainBundle: Bundle.framework)
+  }
+
+  func testLive_BackedProject_BackedReward() {
+    combos([Language.en], [Device.phone4_7inch], self.allRewards()).forEach { language, device, rewardTuple in
+      withEnvironment(language: language) {
+        let (rewardDescription, reward) = rewardTuple
+
+        let project = Project.cosmicSurgery
+          |> Project.lens.state .~ .live
+          |> Project.lens.personalization.isBacking .~ true
+          |> Project.lens.personalization.backing .~ (
+            .template
+              |> Backing.lens.reward .~ reward
+              |> Backing.lens.rewardId .~ reward.id
+              |> Backing.lens.shippingAmount .~ 10
+              |> Backing.lens.amount .~ 700
+          )
+
+        let vc = rewardCardInViewController(
+          language: language,
+          device: device,
+          project: project,
+          reward: reward
+        )
+
+        FBSnapshotVerifyView(vc.view, identifier: "\(rewardDescription)_lang_\(language)_device_\(device)")
+      }
+    }
+  }
+
+  func testLive_BackedProject_NonBackedReward() {
+    combos([Language.en], [Device.phone4_7inch], self.allRewards()).forEach { language, device, rewardTuple in
+      withEnvironment(language: language) {
+        let (rewardDescription, reward) = rewardTuple
+
+        let project = Project.cosmicSurgery
+          |> Project.lens.state .~ .live
+          |> Project.lens.personalization.isBacking .~ true
+          |> Project.lens.personalization.backing .~ (
+            .template
+              |> Backing.lens.reward .~ Reward.otherReward
+              |> Backing.lens.rewardId .~ Reward.otherReward.id
+              |> Backing.lens.shippingAmount .~ 10
+              |> Backing.lens.amount .~ 700
+          )
+
+        let vc = rewardCardInViewController(
+          language: language,
+          device: device,
+          project: project,
+          reward: reward
+        )
+
+        FBSnapshotVerifyView(vc.view, identifier: "\(rewardDescription)_lang_\(language)_device_\(device)")
+      }
+    }
+  }
+
+  func testLive_NonBackedProject() {
```

Comment:
One thing to note is that there are actually two different kinds of "live, non-backed project" states - one when the user is logged in, and one when the user is logged out. The difference between these two from the data perspective is the value of `project.personalization.isBacking` and `project.personalization.backing`. If the user is logged out, both of these will be `nil`. If the user is logged in `project.personalization.isBacking` will be `false`, and `project.personalization.backing` will be `nil`.

It might be worthwhile to add a test for the logged-out state.

Reply:
Ah thanks will do.